src/poetry/inspection/info.py

```diff
@@ -542,7 +543,33 @@ def get_pep517_metadata(path: Path) -> PackageInfo:
             info = PackageInfo.from_metadata_directory(dest)
         except BuildBackendException as e:
             logger.debug("PEP517 build failed: %s", e)
-            raise PackageInfoError(path, e, "PEP517 build failed")
+
+            if isinstance(e.exception, subprocess.CalledProcessError):
+                inner_traceback = (
+                    e.exception.output.decode()
+                    if type(e.exception.output) is bytes
+                    else e.exception.output
+                )
+                inner_reason = "\n    | ".join(
+                    ["", str(e.exception), "", *inner_traceback.split("\n")]
+                )
+                reasons = [
+                    f"<warning>{inner_reason}</warning>",
+                    (
+                        "<info>"
+                        "<options=bold>Note:</> This error originates from the build backend, and is likely not a "
+                        f"problem with poetry but with the package at {path}\n\n"
+                        "  (a) not supporting PEP 517 builds\n"
+                        "  (b) not specifying PEP 517 build requirements correctly; or\n"
+                        "  (c) the build requirement not being successfully installed in your system environment.\n\n"
```

Comment:
```suggestion
                        "<options=bold>Note:</> This error originates from the build backend, and is likely not a "
                        f"problem with poetry but one of the following issues with the package at {path}\n\n"
                        "  - not supporting PEP 517 builds\n"
                        "  - not specifying PEP 517 build requirements correctly; or\n"
                        "  - the build requirements not being successfully installed in your system environment.\n\n"
```

Not sure about the `system environment` part. For legacy reasons we sometimes call the environment where Poetry itself is installed a "system environment", so might be confusing. That could also suggest that builds are not isolated, but using the Python installation global env for builds. All that being said, I don't know how to rewrite that part.

Reply:
How about something like this?

"the build requirements not being successfully installed due to incompatible operating system versions and/or dependencies (eg: compilers, system libraries, header files etc.)"

Reply:
It's a bit verbose but sounds good to me. Any chance you could demo it and post a gif/screenshot of the formatted text? I would like to see how the formatting and colors look.

Reply:
@Secrus made changes and also linked an asciinema url in the description.

Reply:
Thanks. Looks great